### PR TITLE
shipit_code_coverage_crawler: Create coverage-crawler project

### PR DESCRIPTION
**Pull request by rhcu**
- [x] Cleanup template, removing the backend stuff

### CODE REVIEW

File: src/shipit_code_coverage_crawler/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ releng_pkgs
+}: 
+
+let
+
+  inherit (releng_pkgs.lib) mkBackend fromRequirementsFile filterSource;
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/default.nix]: We have to replace mkBackend with mkBot (example usage in src/shipit_code_coverage/default.nix).

File: src/shipit_code_coverage_crawler/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    buildInputs =
+      fromRequirementsFile ./requirements-dev.txt python.packages;
+    propagatedBuildInputs =
+      fromRequirementsFile ./requirements.txt python.packages;
+    passthru = {
+      update = writeScript "update-${name}" ''
+        pushd ${self.src_path}
+        ${pypi2nix}/bin/pypi2nix -v \
+          -V 3.6 \
+          -E "postgresql" \
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/default.nix]: We won't need postgresql

File: src/shipit_code_coverage_crawler/requirements-dev.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+-r ./../../lib/backend_common/requirements-dev.txt
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/requirements-dev.txt]: Replace 'backend_common' with 'cli_common'

File: src/shipit_code_coverage_crawler/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+-e ./../../lib/cli_common[log] #egg=mozilla-cli-common
+-e ./../../lib/backend_common[log,security,cors,api,auth,db] #egg=mozilla-backend-common
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/requirements.txt]: Remove this line

File: src/shipit_code_coverage_crawler/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+
+psycopg2
+gunicorn
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/requirements.txt]: Remove these two requirements

File: src/shipit_code_coverage_crawler/settings.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    taskcluster_client_id=os.environ.get('TASKCLUSTER_CLIENT_ID'),
+    taskcluster_access_token=os.environ.get('TASKCLUSTER_ACCESS_TOKEN'),
+)
+
+locals().update(secrets)
+
+SECRET_KEY = base64.b64decode(secrets['SECRET_KEY_BASE64'])
+
+
+# -- DATABASE -----------------------------------------------------------------
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/settings.py]: Remove this section

File: src/shipit_code_coverage_crawler/settings.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+
+DEBUG = bool(os.environ.get('DEBUG', False))
+
+
+# -- LOAD SECRETS -------------------------------------------------------------
+
+required = [
+    'SECRET_KEY_BASE64',
+    'DATABASE_URL',
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/settings.py]: Remove 'DATABASE_URL'.

File: src/shipit_code_coverage_crawler/setup.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+                pass
+            else:
+                lines.append(line)
+    return lines
+
+
+setup(
+    name='mozilla-shipit-code-coverage-crawler',
+    version=version,
+    description='The code behind code-coverage-crawler.shipit.mozilla-releng.net',
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/setup.py]: Replace the description with the actual description of the project

File: src/shipit_code_coverage_crawler/setup.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    author='Mozilla Release Services Team',
+    author_email='release-services@mozilla.com',
+    url='code-coverage-crawler.shipit.mozilla-releng.net',
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/setup.py]: We'll have to replace the URL too when we have the github repo where we'll upload the reports

File: src/shipit_code_coverage_crawler/shipit_code_coverage_crawler/__init__.py [hunk 10]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/shipit_code_coverage_crawler/__init__.py]: Remove everything from this file except these comments

File: src/shipit_code_coverage_crawler/shipit_code_coverage_crawler/api.py [hunk 11]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/shipit_code_coverage_crawler/api.py]: Remove this file

File: src/shipit_code_coverage_crawler/shipit_code_coverage_crawler/api.yml [hunk 12]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/shipit_code_coverage_crawler/api.yml]: Remove this file

File: src/shipit_code_coverage_crawler/shipit_code_coverage_crawler/flask.py [hunk 13]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/shipit_code_coverage_crawler/flask.py]: Remove this file

File: src/shipit_code_coverage_crawler/tests/conftest.py [hunk 14]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from __future__ import absolute_import
+
+import pytest
+
+import backend_common
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/tests/conftest.py]: Remove everything from here to the end

**Comment by marco-c** [comment]
Don't hesitate to ping me on IRC or email or adding comments here, I know the setup of a new project and of the dev environment for mozilla/release-services is not easy!

File: src/shipit_code_coverage_crawler/tests/test_example.py [hunk 15]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+
+
+def test_example(app):
+    assert True
```
Comment (rhcu) [inline src/shipit_code_coverage_crawler/tests/test_example.py]: I also need to remove this file, yes?

Comment (marco-c) [reply src/shipit_code_coverage_crawler/tests/test_example.py]: This is a placeholder test, but we can keep it and then modify it when we'll actually write tests.

File: src/shipit_code_coverage_crawler/setup.py (hunk 9 above)
Comment (rhcu) [reply src/shipit_code_coverage_crawler/setup.py]: Left it unchanged at this point.

File: src/shipit_code_coverage_crawler/tests/conftest.py [hunk 16]
```diff
@@ -4,24 +4,3 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 from __future__ import absolute_import
-
-import pytest
```
Comment (rhcu) [inline src/shipit_code_coverage_crawler/tests/conftest.py]: Removed `pytest` import too because of flake8.

**Comment by marco-c** [comment]
Since you removed some things from requirements.txt, you will need to run:
`./please tools update-dependencies shipit-code-coverage-crawler`
to automatically update requirements_frozen.txt and requirements.nix.

**Comment by marco-c** [comment]
Let's merge the skeleton first, I'll file issues to implement the actual features.

**Comment by rhcu** [comment]
I have taskcluster test failing - should I resolve this first?

**Comment by garbas** [comment]
@rhcu the taskcluster error you are seeing is due to the repository change that we recently do "mozilla-releng/services" -> "mozilla/release-services". Fix for this is in master. Can you rebase you changes on current master? If you need help with this, let me know I can also do it for you.

**Comment by rhcu** [comment]
Thanks for an explanation. I've rebased but the check is still failing.

**Comment by marco-c** [comment]
Now it is a proper error, you can see it if you click on "Details" on the right of "Taskcluster", then you select the task, then you select "Run logs".

The error is `please_cli.check_cache: [Errno 2] No such file or directory: "error: attribute 'deploy' in selection path 'shipit-code-coverage-crawler.deploy.testing' not found"`

File: src/shipit_code_coverage_crawler/default.nix [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+
+  self = mkPython {
+    inherit python name dirname;
+    version = fileContents ./VERSION;
+    src = filterSource ./. { inherit name; };
+    buildInputs =
+      fromRequirementsFile ./requirements-dev.txt python.packages;
+    propagatedBuildInputs =
+      fromRequirementsFile ./requirements.txt python.packages;
+    passthru = {
```
Comment (marco-c) [inline src/shipit_code_coverage_crawler/default.nix]: The error is caused by the fact that `deploy` is missing here, take a look at the shipit-code-coverage case: https://github.com/mozilla/release-services/blob/54e4b57aefc0f28c69f4ce363edf447379d4c66f/src/shipit_code_coverage/default.nix#L143.

Comment (rhcu) [reply src/shipit_code_coverage_crawler/default.nix]: I found this error: was waiting for please check to be finished before committing. Thanks for helping.

**Pull request merged**